Applications/create-sdm.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Medical Image Registration ToolKit (MIRTK)
+ *
+ * Copyright 2008-2015 Imperial College London
+ * Copyright 2008-2013 Daniel Rueckert, Julia Schnabel
+ * Copyright 2013-2015 Andreas Schuh
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mirtk/Common.h"
+#include "mirtk/Options.h"
+
+#include "mirtk/BaseImage.h"
+#include "mirtk/ImageReader.h"
+#include "mirtk/IOConfig.h"
+#include "mirtk/Transformation.h"
+#include "mirtk/FreeFormTransformation.h"
+#include "mirtk/MultiLevelTransformation.h"
+#include "mirtk/BSplineFreeFormTransformationStatistical.h"
+
+using namespace mirtk;
+
+// =============================================================================
+// Help
+// =============================================================================
+
+// -----------------------------------------------------------------------------
+void PrintHelp(const char *name)
+{
+  cout << "\n";
+  cout << "Usage: " << name << "<matrix> <vector> <dofin> <output>\n";
+  cout << "\n";
+  cout << "Description:\n";
+  cout << "  Creates a statistical deformation model (SDM) file to be used in an\n";
+  cout << "  SDM registration procedure.\n";
+  cout << "\n";
+  cout << "Arguments:\n";
+  cout << "  matrix   File containing a mirtkMatrix whose columns are the eigenvectors of the statistical model.\n";
```

Comment:
Note that when MIRTK was built WITH_MATLAB, these input files could also be ```.mat``` files. Then you don't need to convert from MATLAB to MIRTK file format when your training is done in MATLAB. Otherwise, when the file name extension of these files is not ```.mat```, assume they are in binary MIRTK Matrix/Vector format.